mapie/futur/split/regression.py

```diff
@@ -0,0 +1,231 @@
+        self.cv = cv
+        self.predictor = predictor
+        self.conformity_score = conformity_score
+        self.calibrator = calibrator
+        self.alpha = alpha
+
+    def _check_fit_parameters(self) -> RegressorMixin:
+        """
+        Check and replace default value of ``predictor`` and ``cv`` arguments.
+        Copy the ``predictor`` in ``predictor_`` attribute if ``cv="prefit"``.
+        """
+        self.cv = self._check_cv(self.cv)
+        predictor = check_estimator_regression(self.predictor, self.cv)
+        return predictor
+
+    def _check_calibrate_parameters(self) -> Tuple[
+        ConformityScore, BaseCalibrator
+    ]:
+        """
+        Check and replace default ``conformity_score``, ``alpha`` and
+        ``calibrator`` arguments.
+        """
+        conformity_score_ = check_conformity_score(
+            self.conformity_score, self.default_sym_
+        )
+        calibrator = check_calibrator(self.calibrator)
+        self._check_alpha(self.alpha)
+        calibrator.sym = conformity_score_.sym
+        calibrator.alpha = self.alpha
+        calibrator.random_state = self.random_state
+        return conformity_score_, calibrator
+
+    def predict_score(
+        self, X: ArrayLike
+    ) -> NDArray:
+        """
+        Compute the predictor prediction, used to compute the
+        conformity scores.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed values.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, )
+            predictions
+        """
+        return self.predictor_.predict(X)
+
+    def predict_bounds(
+        self,
+        X: ArrayLike,
+        y_pred: NDArray,
+        **kwargs,
+    ) -> NDArray:
+        """
+        Compute the bounds, using the fitted ``_calibrator``.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed values.
+
+        y_pred: 2D NDArray
+            Observed Target
+
+        z: ArrayLike
+            Exogenous variables
+
+        Returns
+        -------
+        NDArray
+            Bounds, as a 3D array of shape (n_samples, 2, 1)
+            (because we only have 1 alpha value)
+        """
+        predict_kwargs = self.get_method_arguments(
+            self.calibrator_.predict,
+            dict(zip(["X", "y_pred"], [X, y_pred])),
+            kwargs,
+        )
+        conformity_score_pred = self.calibrator_.predict(**predict_kwargs)
+
+        y_pred_low = self.conformity_score_.get_estimation_distribution(
+            X, y_pred[:, np.newaxis], conformity_score_pred[:, [0]]
+        )
+        y_pred_up = self.conformity_score_.get_estimation_distribution(
+            X, y_pred[:, np.newaxis], conformity_score_pred[:, [1]]
+        )
+
+        check_lower_upper_bounds(y_pred_low, y_pred_up, y_pred)
+
+        return np.stack([y_pred_low, y_pred_up], axis=1)
+
+    def predict_best(self, y_pred: NDArray) -> NDArray:
```

Comment:
What is this?

Reply:
Okay, fair enough. Is it really the best that is used in classification though?

Reply:
It is the class with the biggest proba. Maybe the naming can be improved.